src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/base.py

```diff
@@ -188,28 +188,30 @@ def _validate_operational_conditions(
                     f" The affected time steps will not be calculated, and rate is set to zero."
                 )
                 rate = np.where(intermediate_pressure <= 0, 0, rate)
-            intermediate_pressure = np.where(
-                np.logical_and(np.min(rate, axis=0) <= 0, intermediate_pressure <= 0), 1, intermediate_pressure
-            )
+                intermediate_pressure = np.where(intermediate_pressure <= 0, 1, intermediate_pressure)
         if not np.all(suction_pressure > 0):
             logger.warning(
                 f"Inlet pressure needs to be a positive value. Given values: {suction_pressure.tolist()}."
                 f" The affected time steps will not be calculated, and rate is set to zero."
             )
             rate = np.where(suction_pressure <= 0, 0, rate)
-        suction_pressure = np.where(
-            np.logical_and(np.min(rate, axis=0) <= 0, suction_pressure <= 0), 1, suction_pressure
-        )
+            suction_pressure = np.where(suction_pressure <= 0, 1, suction_pressure)
         if not np.all(discharge_pressure > 0):
             logger.warning(
                 f"Outlet pressure needs to be a positive value. Given values: {discharge_pressure.tolist()}"
                 f" The affected time steps will not be calculated, and rate is set to zero."
             )
             rate = np.where(discharge_pressure <= 0, 0, rate)
-        discharge_pressure = np.where(
-            np.logical_and(np.min(rate, axis=0) <= 0, discharge_pressure <= 0), 1, discharge_pressure
-        )
-
+            discharge_pressure = np.where(discharge_pressure <= 0, 1, discharge_pressure)
+        if not np.all(discharge_pressure >= suction_pressure):
+            logger.warning(
+                f"Inlet pressure needs to be a less than or equal to outlet pressure. Given values for inlet"
+                f" pressure: {suction_pressure.tolist()}. Given values for outlet pressure:"
+                f" {discharge_pressure.tolist()}. The affected time steps will not be calculated,"
+                f" and rate is set to zero."
+            )
+            rate = np.where(discharge_pressure < suction_pressure, 0, rate)
+            suction_pressure = np.where(suction_pressure > discharge_pressure, discharge_pressure, suction_pressure)
```

Comment:
Why do we change suction pressure to discharge pressure here when it is not calculated (ie suction > discharge)? Will it not be confusing?